apps/backend/src/auth/auth.service.ts

```diff
@@ -0,0 +1,35 @@
+import { AuthSchProfile } from '@kir-dev/passport-authsch';
+import { Injectable } from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { User } from '@prisma/client';
+import { PrismaService } from 'nestjs-prisma';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private prisma: PrismaService,
+    private jwtService: JwtService
+  ) {}
+
+  async findOrCreateUser(userProfile: AuthSchProfile): Promise<User> {
+    const user = await this.prisma.user.findUnique({ where: { authSchId: userProfile.authSchId } });
+    if (user) return user;
+    return await this.prisma.user.create({
+      data: {
+        authSchId: userProfile.authSchId,
+        fullName: `${userProfile.lastName} ${userProfile.firstName}`, //this is kinda cring...
```

Comment:
Im not sure, how this handles third names here...

Reply:
the displayName scope and property from authSch is actually the full name. I thought it was the authsch username, but no. Might be worth renaming it in the package